src/awscli_login/__main__.py

```diff
@@ -77,29 +81,31 @@ def daemonize(profile: Profile, session: Session, client: boto3.client,
             logger = configFileLogger(profile.logfile, logging.INFO)
             logger.info('Startig refresh process for role %s' % role[1])
 
-            # TODO add retries!
             while(True):
-                retries = 0
-                nap(expires, 0.9, profile.refresh)
-
-                while(True):
-                    try:
-                        saml, _ = refresh(
-                            profile.ecp_endpoint_url,
-                            profile.cookies,
-                        )
-                    except Exception as e:
-                        retries += 1
-
-                        if (retries < 4):
-                            logger.info('Refresh failed: %s' % str(e))
-                            nap(expires, 0.2, profile.refresh * 0.2)
+                try:
+                    retries = 0
+                    nap(expires, 0.9, profile.refresh)
+
+                    while(True):
+                        try:
+                            saml, _ = refresh(
+                                profile.ecp_endpoint_url,
+                                profile.cookies,
+                            )
+                        except Exception as e:
+                            retries += 1
+
+                            if (retries < 4):
+                                logger.info('Refresh failed: %s' % str(e))
+                                nap(expires, 0.2, profile.refresh * 0.2)
+                            else:
+                                raise
                         else:
-                            raise
-                    else:
-                        break
+                            break
 
-                expires = save_sts_token(session, client, saml, role)
+                    expires = save_sts_token(session, client, saml, role)
+                except SIGINT:
```

Comment:
Not a big deal, but you could omit the `except SIGINT:` if you're not doing anything with it here.

Reply:
The SIGINT has a use. It is used to goose the process into doing another refresh. When a SIGINT is received an exception is thrown. It needs to be caught, but a pass is enough to cause another loop and refresh. I guess I should have added a comment on that fact...

Reply:
I agree with the idea of a comment, since this is perhaps a bit subtle, and someone might wind up tripping over it later.